python/lsst/dax/apdb/apdb.py

```diff
@@ -180,7 +180,8 @@ def tableDef(self, table: ApdbTables) -> Table | None:
     def getDiaObjects(self, region: Region) -> pandas.DataFrame:
         """Return catalog of DiaObject instances from a given region.
 
-        This method returns only the last version of each DiaObject. Some
+        This method returns only the last version of each DiaObject, and may
+        contain only a subset of the DiaObject columns. Some
```

Comment:
I'd say something about the subset: it is the subset of columns required for AP association, and you get a subset unless the database has not been optimized.  As a result, this method is not suitable for DiaObject retrieval for general analysis purposes.

Reply:
I'll defer that to @andy-slac; while we didn't talk about this particular point, in our previous discussions, he tended to ask for specs to be as open-ended as possible.

Reply:
`Apdb` interface is not meant for general analysis, only for AP pipeline use. But it's OK to say that subset must satisfy AP association needs, that is certainly what it has to do.